### PR TITLE
CMR-6287: Changes to structure of science keyword indexing and response

### PR DESCRIPTION
**Pull request by mschmele**
As requested by EDSC

### CODE REVIEW

File: indexer-app/src/cmr/indexer/services/index_service.clj [hunk 1]
```diff
@@ -176,17 +177,57 @@
...
+                             first
+                             (nth keyword-hierarchy)
+                             name
+                             csk/->snake_case))
+        keyword-string (->> keyword-hierarchy
+                            (map #(get science-keywords %))
+                            (remove nil?)
+                            (s/join ":"))]
+     {:type "science_keywords"
+      :_id (str keyword-string "_science_keywords")
```
Comment (jaybarra) [inline indexer-app/src/cmr/indexer/services/index_service.clj:211]: do any science keywords contain special characters?

File: system-int-test/test/cmr/system_int_test/search/autocomplete_collection_facet_search_test.clj [hunk 2]
```diff
@@ -35,7 +39,6 @@
     (doseq [doc documents] (debug "ingested " doc))
     (index/wait-until-indexed)
     (f)
-    (doseq [doc documents] (esd/delete conn "1_autocomplete" "suggestion" (:_id doc)))
     (index/wait-until-indexed)))
```
Comment (jaybarra) [inline system-int-test/test/cmr/system_int_test/search/autocomplete_collection_facet_search_test.clj:42]: can remove this if we are no longer cleaning up the created documents. Can tests be run consecutively without this cleanup happening i.e. the counts remain correct?

File: indexer-app/src/cmr/indexer/services/index_service.clj [hunk 3]
```diff
@@ -1,6 +1,7 @@
 (ns cmr.indexer.services.index-service
   "Provide functions to index concept"
   (:require
+   [camel-snake-kebab.core :as csk]
```
Comment (CayvonH) [inline indexer-app/src/cmr/indexer/services/index_service.clj:4]: nit: no more shorthand namespace requires.

File: indexer-app/src/cmr/indexer/services/index_service.clj [hunk 4]
```diff
@@ -176,18 +177,52 @@
...
+        terminal-key (->> keyword-hierarchy
+                          (map #(get science-keywords %))
+                          (positions nil?))
+        keyword-string (->> keyword-hierarchy
+                            (map #(get science-keywords %))
+                            (remove nil?)
+                            (s/join ":"))
+        keyword-value (get science-keywords terminal-key)]
+     {:type "science_keywords"
+      :_id (str (s/lower-case keyword-string) "_science_keywords")
```
Comment (CayvonH) [inline indexer-app/src/cmr/indexer/services/index_service.clj:202]: How big can this string be? `_id` is limited to 512 bytes.

Comment (mschmele) [reply indexer-app/src/cmr/indexer/services/index_service.clj:202]: It's never going to be anything too crazy - the values are humanized

Comment (mschmele) [reply indexer-app/src/cmr/indexer/services/index_service.clj]: The following value is 85 bytes before I convert it to a hash and is a pretty typical length for a science keyword
`Biosphere:Ecological Dynamics:Ecosystem Functions:Respiration Rate:Evapotranspiration`

Comment (CayvonH) [reply indexer-app/src/cmr/indexer/services/index_service.clj]: Okay cool - thanks for implementing it anyway :)

**Pull request merged**